lmms_eval/tasks/charxiv/descriptive_utils.py

```
@@ -0,0 +1,208 @@
+    query = ""
+    for i, (_, response, answer) in enumerate(batch):
+        # index, response, answer
+        curr_query = "T{}:\nResponse {}: {}\nGround Truth {}: {}\n\n".format(i + 1, i + 1, response, i + 1, answer)
+        query += curr_query
+    return query
+
+
+def verify_grading_output(data, length_data):
+    # check the integrity of keys and values
+    for i in range(1, length_data + 1):
+        assert f"extract_answer_T{i}" in data, f"extract_answer_T{i} is not found in {d}"
```

Ruff (0.12.2)

Undefined name `d`

(F821)